File: crates/base/src/model/sequence/action.rs
// Copyright (c) nyanbot.com 2025.
// This file is licensed under the AGPL-3.0-or-later.

use crate::model::{Sequence, TelegramActionButtonConfig};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(tag = "type", rename_all = "SCREAMING_SNAKE_CASE")]
pub enum Action {
	AndThen { action: Box<Action>, sequence: Box<Sequence> },
	Buy,
	NotifyTelegram {
		buttons: Vec<TelegramActionButtonConfig>
	},
	Sell,
}

File: bin/frontend/src/types/rules/action.ts
import {ValueNumber} from "@types";

export enum ActionType {
    NOTIFY_TELEGRAM = 'NOTIFY_TELEGRAM'
}

export enum NotificationChannel {
    TELEGRAM = 'TELEGRAM'
}

export type TelegramButtonAction = 'NOTHING' | 'BUY' | 'SELL'

export type TelegramButtonConfig = {
    action: TelegramButtonAction;
    value?: ValueNumber;
}

export type Action = ActionNotifyTelegram;

export type ActionNotifyTelegram = {
    type: ActionType,
    buttons: Array<TelegramButtonConfig>,
}


